### ISSUE TITLE
Lerna seems to not found packages after upgrade

### ISSUE THREAD

**Issue by dboulardsc** [lerna/lerna#4056]
## Current Behavior
Updated from v7 to v8 with the following config
```json
{
  "$schema": "node_modules/lerna/schemas/lerna-schema.json",
  "version": "1.0.0",
  "npmClient": "yarn"
}
```

Executing list returns to have found 0 packages.
Meanwhile :
* Occurs only on this very computer. My other one is okay with the upgrade. Also ok for every other colleague in my team. Also ok for github ci/cd.
* `yarn workspace info` returns the correct  amount of packages
* `npx lerna clean` seems to find the various packages to remove the node_modules
* Tried on a new-from-scratch project and same behaviour.
* Manually adding packages in lerna.json or root package.json doesnt change anything
* `yarn cache clean` doesnt change anything
* `rm ~/.yarn` doesnt change anything

Also, seems quite linked to https://github.com/lerna/lerna/issues/4018

## Expected Behavior

List the packages my monorepo contains.

## Steps to Reproduce

Started a project from scratch and can reproduce the issue.
* yarn init
* yarn add lerna
* lerna init
* mkdir packages
* lerna create patata
* lerna create patato

root package.json :
```json
{
  "private": true,
  "workspaces": [
    "packages/*"
  ],
  "name": "lernatest",
  "version": "1.0.0",
  "main": "index.js",
  "license": "MIT",
  "dependencies": {
    "lerna": "^8.1.7"
  }
}
```

first package :
```json
{
  "name": "patata",
  "version": "1.0.0",
  "main": "lib/patata.js",
  "scripts": {
    "build": "echo \"build patata\""
  },
  "dependencies": {
    "patato": "1.0.0"
  }
}
```

second package :
```json
{
  "name": "patato",
  "version": "1.0.0",
  "main": "lib/patato.js",
  "scripts": 
    "build": "echo \"build patato\""
  }
}
```

lerna.json :
```json
{
  "$schema": "node_modules/lerna/schemas/lerna-schema.json",
  "version": "1.0.0",
  "npmClient": "yarn"
}
```

### Failure Logs / Configuration

<details><summary>lerna.json</summary><p>

```json
<!-- Please paste your `lerna.json` here -->
```

### Environment

```
lerna notice cli v8.1.7

 Environment info:

  System:
    OS: Linux 5.15 Manjaro Linux
    CPU: (8) x64 11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz
  Binaries:
    Node: 20.16.0 - ~/.nvm/versions/node/v20.16.0/bin/node
    Yarn: 1.22.22 - /usr/bin/yarn
    npm: 10.8.1 - ~/.nvm/versions/node/v20.16.0/bin/npm
  Utilities:
    Git: 2.45.2 - /usr/bin/git
  npmPackages:
    lerna: ^8.1.7 => 8.1.7 
```

**Comment by JamesHenry** [comment]
@dboulardsc Sorry to hear that, issues that only happen under some niche circumstances are the worst, because I naturally can't repro either:

![image](https://github.com/user-attachments/assets/929d4a64-4166-493a-9d35-094b20a36027)

Some ideas to try (in order from simplest to most effort, not necessarily most likely to be the reason):

- Running `npm ls nx` to check you have 19.5.6 or similar installed and not an old version
- Running `yarn nx reset` and retrying the commands
- Is the directory you are working in nested under another git directory? It's unusual and almost always accidental, but I believe if a parent directory had a .git directory it could cause weird issues (and that would be the kind of thing that could happen on one specific user's machine). As well as just trying to find .git directories, in general, please try just moving the lerna workspace to a different location on you machine and see if it changes anything.
- Uninstalling and reinstalling `yarn` classic altogether

**Comment by GregoireBellon** [comment]
I face the same issue with npm and lerna 8.1.8. 

I have two packages (@package/front and @package/back) that both share a dependency (@package/shared)

```npx nx graph``` shows the dependency relationship, but the command ```lerna run build --scope @package/back--include-dependencies``` does not build my shared dependency. 

The output is : 
```
lerna notice cli v8.1.8
lerna notice filter including "@package/back"
lerna notice filter including dependencies
lerna info filter [ '@package/back' ]

> @package/back:build

> build
> nest build
... then fails irremediably
```

The package.json files are very similar to @dboulardsc  's ones, but the dependencies are declared like this : 

```
  "dependencies": {
    "@package/shared": "*"
  }
```

The command used to work, and no package.json of the monorepo changed significantly, so I guess that the issue could have appeared in lerna 8.1.8 ?

**Comment by loganzartman** [comment]
same issue here--showing zero packages.
I'm defining packages via yarn classic `workspaces`. I'm stuck on yarn 1.18.0 (sadly).

I agree with @ArthurG94: I decremented the lerna version and tested each one. It starts working again at 8.1.2. I would consider this a regression between 8.1.2 and 8.1.3. This is the only thing I changed, and I was able to repro this consistently when changing versions. I have only the minimal bootstrapped lerna config:

```
{
  "$schema": "node_modules/lerna/schemas/lerna-schema.json",
  "version": "0.0.0",
  "npmClient": "yarn"
}
```

**Comment by kaiz-io** [comment]
So I found that I had I am using dotfiles that has a `~/.gitignore` which has a * and then I add ! for files that I want my dotfiles to track. This does not affect my any other git repo commands tracking only lerna is affected by this. 

```
~ 
|>.gitignore                                                      < this .gitignore has an affect on lerna
|_code
|_____|_playground
|__________________|_lerna                             < directory where I have my simple lerna project
__________________________|_lerna.json
__________________________|_.gitignore          < standard
```

~/.gitingore
```
*
!.gitignore
.gitconfig
!code/playground/lerna/* < have to add this line for lerna to work.
....
```

~/code/playground/lerna/.gitignore
```
node_modules/
```

**Comment by JamesHenry** [comment]
Thank you @kaiz-io I have a feeling that the `.gitignore` is going to be the explanation in most if not all cases.

If you `.gitignore` a package, then it will be excluded from the project graph construction. This is powered by `nx` core behind the scenes (to allow for free integrations with Nx Cloud and Nx Console, and benefit from all that work within Lerna). There is a `.nxignore` file that has higher precedence than `.gitignore`, so if you want to keep the package git ignored, you can add a `.nxignore` file with the negative pattern in instead:

Please let me know if that works for you.

Please could @loganzartman @GregoireBellon @ArthurG94 @dboulardsc confirm if they also have a git ignore involved and if the negative pattern in `.nxignore` (i.e. forcing it to be considered, despite being git ignored) works?

**Comment by loganzartman** [comment]
I'll get back to you when I have time to check the fix, but I'll say that I do have a .gitignore in my home directory. So:

```
/home/logan
    .git
    .gitignore
    project
        .git
        .gitignore
```

However there is no reason the first .gitignore should apply to nx, because the lerna project itself is in a git repo. Git does not work the way that nx is here, otherwise this setup would not work and no one would use it. So maybe this is a bug in nx rather than lerna, but I don't think it's user error.

**Comment by ArthurG94** [comment]
Hi,
I can't check the fix either, but I confirm that I have a .gitignore file in my home folder that excludes everything except a few config files.

**Comment by JamesHenry** [comment]
@loganzartman Thanks a lot for that overview, I was able to reproduce and we are working on a fix

**Comment by loganzartman** [comment]
Appreciate the info @JamesHenry, is there an issue or PR to track for the fix? I assume on the nx side?

**Comment by JamesHenry** [comment]
Yes, because it's not hard blocked (the ignore file usage works) it's going into our next cooldown cycle, rest assured I will post here when it's available

**Comment by loganzartman** [comment]
for anyone else with my issue (/home/logan/.gitignore), I upgraded to nx@20 and added:

.nxignore
```
!*
```

I don't know yet if there's any disadvantage to brute-forcing this instead of specifically un-excluding my packages, but this gets the project graph working at least.
